plugin/src/main/java/dev/regadas/trino/pubsub/listener/PubSubEventListener.java

```diff
@@ -42,33 +42,33 @@ public static final PubSubEventListener create(
         return create(config, publisher, stats);
     }
 
-    public static final PubSubEventListener create(
```

Comment:
why did we end up removing final here?

Reply:
The `static`methods cannot be overridden so `final` modifier is no use here. IntelliJ highlighted it that it was "superfluous" so I removed it.

Technically [there is](https://coderanch.com/wiki/659959/Overriding-Hiding) a rare use for `final` on `static` methods but is so corner-case that I prefer a less cluttered code.

Reply:
actually in this case maybe Intellij hint was because this was an interface. Now that is a class it's different.

Reply:
> the combo with final static when overriding is not necessary hints the compiler to inline code.
Do you have a link to some docs? To my eyes, static methods are early bound (they are not overridable therefore no dynamic dispatch). I fail to see what `final` will add to that that will make the JVM to in line it.

Reply:
ahah honestly there's a slight chance that I might be mixing things from the "better java" (scala) here. We can easily check the bytecode.

Nevertheless, here doesn't really matter since the class is final making the final in the static method superfulous kinda similar to the interface case.